### PR TITLE
Conditionalize config

### PR DESCRIPTION
**Pull request by shakefu**
- Fixes behavior so that if .pre-commit-config.yaml is missing, it won't fail the check
- Conditionalizes the config param to work if there is no commitlint config
- Handle edge case where executing repo's context.base_ref is different from workspace repo's (e.g. if we're on main, but check is running against a repo with master)
- Add CI tests running against compliant OSS repositories
- Refactor to use commitlint subaction instead of reproducing code

### CODE REVIEW

**Review by brianpham — APPROVED** [review]

File: action.yaml [hunk 1]
```diff
@@ -18,160 +18,187 @@ inputs:
...
+      # TODO: use a released version of this action
+      uses: open-turo/action-pre-commit/commitlint@conditionalize-config
       with:
-        output: " "
-    # Export changed file names for use in pre-commit action
-    - name: Limit pre-commit to changed files
+        restore-config: "false"
+        config-file: ${{ inputs.config-file }}
+        turo-conventional-commit: ${{ inputs.turo-conventional-commit }}
+    # - name: Install @commitlint/cli
```
Comment (yohanb) [inline action.yaml:83]: we could remove the comment block

**Review by yohanb — APPROVED** [review]

File: action.yaml (hunk 1 above)
Comment (shakefu) [reply action.yaml:83]: Yeap wanted to make sure there wasn't an issue running the sub-action beforehand.
